flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/hint/FlinkHints.java

```diff
@@ -189,18 +192,33 @@ public static RelNode capitalizeJoinHints(RelNode root) {
 
     private static class CapitalizeJoinHintShuttle extends RelShuttleImpl {
 
+        @Override
+        public RelNode visit(LogicalCorrelate correlate) {
+            return visitBiRel(correlate);
+        }
+
         @Override
         public RelNode visit(LogicalJoin join) {
-            List<RelHint> hints = join.getHints();
+            return visitBiRel(join);
+        }
+
+        public RelNode visitBiRel(BiRel biRel) {
+            Hintable hBiRel = (Hintable) biRel;
             AtomicBoolean changed = new AtomicBoolean(false);
             List<RelHint> hintsWithCapitalJoinHints =
-                    hints.stream()
+                    hBiRel.getHints().stream()
                             .map(
                                     hint -> {
                                         String capitalHintName =
                                                 hint.hintName.toUpperCase(Locale.ROOT);
                                         if (JoinStrategy.isJoinStrategy(capitalHintName)) {
                                             changed.set(true);
+                                            if (hint.listOptions.isEmpty()) {
```

Comment:
We'd better to distinguish the kvOption and listOption via different hint but not the option itself, here we can use `isLookupHint` to determine